internal/webhook/sentinel/handler.go

```diff
@@ -62,67 +62,40 @@ func (h *Handler) Handle(ctx context.Context, req admission.Request) admission.R
 		return admission.Allowed(fmt.Sprintf("operation %s is allowed", req.Operation))
 	}
 
-	requestGK := schema.GroupKind{Group: req.Kind.Group, Kind: req.Kind.Kind}
-
-	obj, err := h.decodeRequestObject(req, requestGK)
+	etcd, allowedMessage, err := h.getEtcdForRequest(ctx, req)
```

Comment:
Please retrieve the etcd object with metadata only, rather than fetching the full object. Since we're not utilizing the full object in this context

Reply:
We are using methods like `etcd.IsDeletionInProgress()` and `etcd.IsReconciliationInProgress()`, which require the status to be present in the object. So we can't use `PartialObjectMetadata` here.